add more examples of line wrapping (#95)
redhat-cop/automation-good-practices: coding_style/README.adoc

      Duis a neque fermentum nulla imperdiet blandit.
----
+

> TIP: Always use the sign `>-` instead of `>` unless you are absolutely sure the trailing newline is not significant.

Reviewer: ```suggestion
TIP: Always use the sign `>-` instead of `>` unless you are absolutely sure the trailing newline is not significant.
The sing `>` adds a newline character to the last line, effecively turning `non nunc.` at the end of the example string above into  `"non nunc.\n"`,`>-` doesn't add the newline character.
```

```
====

== How To Wrap Lines

ansible-lint has a pretty short line length, which causes problems if you are trying to use good programming practices by having descriptive variable names, which usually end up being quite long.
Here are some examples of how to deal with line wrapping in common scenarios:

* Jinja expressions can be wrapped

> Within the Jinja expression, whitespace isn't significant, so take advantage of this to wrap lines as readable as possible:

Reviewer: ```suggestion
Within the Jinja expression, whitespace and newline characters aren't significant, so take advantage of this to wrap lines into as readable a form as possible:
```

```
====

== Wrap longer lines of code

ansible-lint has a pretty short line length, which causes problems if you are trying to use good programming practices by having descriptive variable names, which usually end up being quite long.
Here are some examples of how to deal with line wrapping in common scenarios:

> * Jinja expressions can be wrapped
> 
> Within the Jinja expression, whitespace and newline characters aren't significant, so take advantage of this to wrap lines into as readable a form as possible.
> Remember, in a `when`, `that`, `failed_when`, or other such keywords, you can just write Jinja code - you do not need the `"{{ ... }}"`

Reviewer: ```suggestion
* Jinja expressions can be wrapped.
  Within the Jinja expression, whitespace and newline characters aren't significant, so take advantage of this to wrap lines into as readable a form as possible.
  Remember, in a `when`, `that`, `failed_when`, or other such keywords, you can just write Jinja code - you do not need the `"{{ ... }}"`
```

> * Start an expression with '{{' followed by newline if the line will otherwise be too long
> 
> But what if the code is already indented a lot, and the variable I'm assigning to is already very long, and I can't put anything else on the line?
> Just start the assignment on the next line.
> 
> * Use backslash escapes in double quoted strings
> 
> But what if I have a very long string that I cannot use `>-` to wrap because I cannot have extra spaces in the value e.g. like a url value?
> Use a backslash escape in a double quoted string.
> YAML will concatenate the values with no spaces.

Reviewer: ```suggestion
* Start an expression with '{{' followed by newline if the line will otherwise be too long.
  But what if the code is already indented a lot, and the variable I'm assigning to is already very long, and I can't put anything else on the line?
  Just start the assignment on the next line.

* Use backslash escapes in double quoted strings.
  But what if I have a very long string that I cannot use `>-` to wrap because I cannot have extra spaces in the value e.g. like a url value?
  Use a backslash escape in a double quoted string.
  YAML will concatenate the values with no spaces.
```

> +

Reviewer: ```suggestion
```

[%collapsible]
====
Rationale:: Use of whitespace and multi-line indentation makes expressions easier to read.

.Do this:
[source,yaml]
----
- name: Wrap long Jinja expressions
  foo: "{{ a_very.long_variable.name |
    somefilter('with', 'many', 'arguments') |
    another_filter | list }}"
  when: a_very.long_variable.name |
    somefilter('with', 'many', 'arguments') |
    another_filter | list

>                   - name: Wrap when first line is already too long
>                     very_indented_foo: "{{
>                       a_very.long_variable.name |
>                       somefilter('with', 'many', 'arguments') |
>                       another_filter | list }}"
>                     when: \
>                       a_very.long_variable.name |
>                       somefilter('with', 'many', 'arguments') |
>                       another_filter | list | length > 0

Reviewer: ```suggestion
- name: Wrap when first line is already too long
  very_indented_foo: "{{
    a_very.long_variable.name |
    somefilter('with', 'many', 'arguments') |
    another_filter | list }}"
  when: \
    a_very.long_variable.name |
    somefilter('with', 'many', 'arguments') |
    another_filter | list | length > 0
```

- name: Set some test variables
  set_fact:
    my_very_long_variable_1: "{{ __pre_digest | filter1 }}"
    my_very_long_variable_2: "{{ __pre_digest | filter2 }}"
  vars:
    __pre_digest: "{{ a_very.long_variable.name | some_filter }}"

- name: Use a very long URL
  uri:
    url: "https://{{ my_very_long_value_for_hostname }}:\
      {{ my_very_long_value_for_port }}\
      {{ my_very_long_value_for_uri }}?\
      {{ my_very_long_value_for_query }}"

----

.Don't do this:
[source,yaml]
----
- name: Very long line with Jinja expression
  foo: "{{ a_very.long_variable.name | somefilter('with', 'many', 'arguments') | another_filter | list }}"

>                   - name: First line is already too long
>                     very_indented_foo: "{{ a_very.long_variable.name |
>                       somefilter('with', 'many', 'arguments') |
>                       another_filter | list }}"
>                     when: a_very.long_variable.name |
>                       somefilter('with', 'many', 'arguments') |
>                       another_filter | list | length > 0

Reviewer: ```suggestion
- name: First line is already too long
  very_indented_foo: "{{ a_very.long_variable.name |
    somefilter('with', 'many', 'arguments') |
    another_filter | list }}"
  when: a_very.long_variable.name |
    somefilter('with', 'many', 'arguments') |
    another_filter | list | length > 0
```
